src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/Runtime/BindingFlagSupport/QueryResult.Enumerator.cs

```diff
@@ -36,14 +36,14 @@ public M Current
                 [MethodImpl(MethodImplOptions.AggressiveInlining)]
                 get
                 {
-                    return _queriedMembers[_index];
+                    return _queriedMembers![_index];
                 }
             }
 
             private int _index;
             private readonly int _unfilteredCount;
             private readonly BindingFlags _bindingAttr;
-            private readonly QueriedMemberList<M> _queriedMembers;
+            private readonly QueriedMemberList<M>? _queriedMembers;
```

Comment:
NIT: might better make non null and assert on constructor

Reply:
The enumerator treats null values as a superficially valid state. I don't see why this should be changed, given it's internal implementation.